File: src/etch/bindings/csharp/transport/Startable.cs
using System;
using System.Text;
using System.Collections.Generic;

namespace Etch.Transport
{
    /// <summary>
    /// Description of Startable
    /// </summary>
    public interface Startable
    {
        /// <summary>
        /// 
        /// </summary>
        /// Exception:
        ///     throws Exception
        void Start();

        /// Exception:
        ///     throws Exception
        void Stop();

        /// <summary>
        /// 
        /// </summary>
        /// <returns>true if started</returns>
        bool IsStarted();

        /// <summary>
        /// Check the startable is started
        /// </summary>
        void CheckIsStarted();
    
    }
}

File: src/etch/util/Startable.java
/* $Id: Startable.java 21882 2006-04-11 18:26:09Z builder $
 *
 * Created by wert on Feb 10, 2006.
 * 
 * Copyright (c) 2007 Metreos, Inc. All rights reserved.
 */

package etch.util;

/**
 * Description of Startable
 */
public interface Startable
{
	/**
	 * @throws Exception
	 */
	public void start() throws Exception;
	
	/**
	 * @throws Exception 
	 */
	public void stop() throws Exception;
	
	/**
	 * @return true if started.
	 */
	public boolean isStarted();

	/**
	 * Checks that the startable is started.
	 * @throws IllegalStateException if not started.
	 */
	public void checkIsStarted();
}
